src/cljs/main/broadfcui/components/entity_details.cljs

```diff
@@ -56,9 +62,12 @@
                                                      :style {:width 120}
                                                      :defaultValue (if redacted-snapshot -1 (key entity))
                                                      :onChange (when-let [f (:onSnapshotIdChange props)]
-                                                                 #(f (int (common/get-trimmed-text refs "snapshotId"))))}
-                   (:snapshots props)
-                   redacted-snapshot)
+                                                                 (case repo
+                                                                   ;; this is silly, but I cannot use let because react prohibits accessing dom nodes in render
```

Comment:
Instead of trying to get the selections here, update state when the selection itself changes and read it from state.